tools/pipelines/test-real-service.yml

```diff
@@ -198,3 +198,31 @@ stages:
                 BUILD_ID: $(Build.BuildId)
                 ADO_API_TOKEN: $(System.AccessToken)
                 PIPELINE: 'EndToEndTests'
+
+            # Retrieve E2E test pass rate
+            - task: Bash@3
+              displayName: Retrieve test pass rate
+              inputs:
+                targetType: 'inline'
+                workingDirectory: $(toolAbsolutePath)
+                script: |
+                  echo: "creating output folder"
+                  mkdir -p ${{ variables.testWorkspace }}/passRateOutput
+                  echo "Executing curl command ..."
+                  echo  'curl -u ":<REDACTED>" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$(Build.BuildId)/timeline"'
+                  curl -u ":$(System.AccessToken)" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$(Build.BuildId)/timeline\?api-version\=6.0-preview.1" > ${{ variables.testWorkspace }}/passRateOutput/output.json
+                  pwd;
+                  ls -laR ${{ variables.testWorkspace }}/passRateOutput/output.json;
+                  stages=($(jq -r '.records[] | select(.type == "Stage") | .identifier' ${{ variables.testWorkspace }}/passRateOutput/output.json))
+                  mkdir -p ${{ variables.testWorkspace }}/stageFiles
+                  for stage in "${stages[@]}"; do
+                    echo "Fetching data for stage: $stage"
+                    curl -s -u ":$(System.AccessToken)" "https://vstmr.dev.azure.com/fluidframework/internal/_apis/testresults/metrics?pipelineId=$(Build.BuildId)&stageName=$(stage)&api-version=7.1-preview.1 > "${{ variables.testWorkspace }}/stageFiles/${stage}.json"
+                    node --require @ff-internal/aria-logger bin/run --handlerModule $(Build.SourcesDirectory)/tools/telemetry-generator/dist/handlers/testPassRate.js --dir '${{ variables.testWorkspace }}/timingOutput/${stage}.json'
```

Comment:
We're outputting the file to `stageFiles/` and then trying to read them from `timingOutput/` (and above we're using `passRateOutput/`). `passRateOutput/` as the single one makes sense, I think. Or we could put the stage files under `passRateOutput/stageFiles/` and then pass that one to the telemetry-generator, so we can call it once instead of once per file (and not have to deal with the `output.json` file from the first step.